wgpu/src/shader/offscreen_blit.wgsl

```diff
@@ -0,0 +1,22 @@
+@group(0) @binding(0) var u_texture: texture_2d<f32>;
+@group(0) @binding(1) var out_texture: texture_storage_2d<rgba8unorm, write>;
+
+fn srgb(color: f32) -> f32 {
+    if (color <= 0.0031308) {
+        return 12.92 * color;
+    } else {
+        return (1.055 * (pow(color, (1.0/2.4)))) - 0.055;
+    }
+}
+
+@compute @workgroup_size(1)
+fn main(@builtin(global_invocation_id) id: vec3<u32>) {
+    // texture coord must be i32 due to a naga bug:
+    // https://github.com/gfx-rs/naga/issues/1997
+    let coords = vec2(i32(id.x), i32(id.y));
+
+    let src: vec4<f32> = textureLoad(u_texture, coords, 0);
+    let srgb_color: vec4<f32> = vec4(srgb(src.x), srgb(src.y), srgb(src.z), src.w);
+
+    textureStore(out_texture, coords, srgb_color);
+}
```

Comment:
This won't work with the `web-colors` feature for similar reasons to #1885.

I think we should avoid doing any color conversions in shader code. These should be handled by the GPU automatically.

I believe we can rewrite this using a render pipeline so we can easily control the output format from Rustland.

Reply:
I will look into it, pretty sure I couldn't use a srgb texture format for a storage texture so it will have to be a render pipeline vs compute, yeah.

Reply:
Not sure I follow. I am fairly certain I have sampled textures with a different format than the output format in a render pipeline.

`glyphon` uses a masked texture with a single channel for most glyphs, for instance.

Reply:
might have a different way to do it using a render pipeline, I'm poking around with it this morning.

Reply:
Nvm i was just being a pepeg, should be good now 👍

Reply:
Taking another look, I think now I can just re-use the existing `blit.wgsl` shader we have for multisampling